pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.js

```diff
@@ -0,0 +1,118 @@
+                if (ERR(err, callback)) return;
+                res.locals.course_instance_stat = result.rows[0];
+                debug(res.locals.course_instance_stat);
+                callback(null);
+            });
+        },
+        (callback) => {
+            debug('query short_names');
+            sqldb.queryOneRow(sql.short_names, {course_id: res.locals.course.id}, (err, result) => {
+                if (ERR(err, callback)) return;
+                res.locals.short_names = result.rows[0].short_names;
+                callback(null);
+            });
+        },
+    ], function(err) {
+        if (ERR(err, next)) return;
+        debug('render page');
+        res.locals.infoCourseInstancePath = encodePath(path.join('courseInstances', res.locals.course_instance.short_name, 'infoCourseInstance.json'));
+        res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+    });
+});
+
+router.post('/', function(req, res, next) {
+    debug('POST /');
+    if (req.body.__action == 'copy_course_instance') {
+        debug('Copy course instance');
+        const editor = new CourseInstanceCopyEditor({
+            locals: res.locals,
+        });
+        editor.canEdit((err) => {
+            if (ERR(err, next)) return;
+            editor.doEdit((err, job_sequence_id) => {
+                if (ERR(err, (e) => logger.error(e))) {
+                    res.redirect(res.locals.urlPrefix + '/edit_error/' + job_sequence_id);
+                } else {
+                    debug(`Get course_instance_id from ciid=${editor.short_name} with course_id=${res.locals.course.id}`);
+                    sqldb.queryOneRow(sql.select_course_instance_id_from_short_name, {short_name: editor.short_name, course_id: res.locals.course.id}, (err, result) => {
+                        if (ERR(err, next)) return;
+                        res.redirect(res.locals.plainUrlPrefix + '/course_instance/' + result.rows[0].course_instance_id + '/instructor/instance_admin/settings');
+                    });
+                }
+            });
+        });
+    } else if (req.body.__action == 'delete_course_instance') {
+        debug('Delete course instance');
+        const editor = new CourseInstanceDeleteEditor({
+            locals: res.locals,
+        });
+        editor.canEdit((err) => {
+            if (ERR(err, next)) return;
+            editor.doEdit((err, job_sequence_id) => {
+                if (ERR(err, (e) => logger.error(e))) {
+                    res.redirect(res.locals.urlPrefix + '/edit_error/' + job_sequence_id);
+                } else {
+                    res.redirect(`${res.locals.plainUrlPrefix}/course/${res.locals.course.id}/course_admin`);
+                }
+            });
+        });
+    } else if (req.body.__action == 'change_id') {
+        debug(`Change short_name from ${res.locals.course_instance.short_name} to ${req.body.id}`);
+        let ciid_new;
```

Comment:
Are we validating `ciid_new` server-side? I can see the code in `canEdit()` that checks whether it's trying to escape the directory hierarchy, but I can't see where we are validating the other constraints (e.g., no spaces). I know we are validating that client-side, but that's only advisory.

(I think this applies to other rename forms as well, but I didn't check them all.)

Reply:
done